agent/engine/docker_image_manager.go

```diff
@@ -335,18 +349,35 @@ func (imageManager *dockerImageManager) deleteImage(imageID string, imageState *
 		seelog.Errorf("Image ID to be deleted is null")
 		return
 	}
-	seelog.Infof("Removing Image: %s", imageID)
-	err := imageManager.client.RemoveImage(imageID, removeImageTimeout)
-	if err != nil {
-		if err.Error() == imageNotFoundForDeletionError {
-			seelog.Errorf("Image already removed from the instance: %v", err)
-		} else {
-			seelog.Errorf("Error removing Image %v - %v", imageID, err)
-			delete(imageManager.imageStatesConsideredForDeletion, imageState.Image.ImageID)
-			return
+
+	// Due to the logic in removeUnusedImages if we don't remove the imagestate
+	// of preserved image, it will always be the LRU image to be deleted, this cause
+	// other image can't be deleted. we can get rid of this after changing the logic
+	// in removeUnusedImages to get all the images to be deleted once or so.
+	preserved := false
+	for _, m := range imageManager.preservedImages {
+		if imageID == m {
+			seelog.Infof("Image %s was marked to be preserved, skipping deletion", imageID)
+			preserved = true
+			break
```

Comment:
Can the `imageState` removal be moved before `RemoveImage`? If so, this could `return` and the `if !preserved` logic below could be removed... Dunno if that is actually clearer though...